imbi/automations.py

```diff
@@ -177,32 +180,43 @@ async def run(self):
                 'gitlab_project_id': gitlab_info['id'],
             })
 
-        if self.settings.get('repository_link_id'):
+        link_id = self.automation_settings['gitlab'].get('repository_link_id')
```

Comment:
I think this should be `link_id = self.settings['gitlab'].get('repository_link_id')` it looks like you removed `automation_settings` in place of passing in the entire app instance which has access to those variables in the settings dict.

Reply:
Also looks like you did the same thing below with the Sonar settings.

Reply:
Looks like another occurrence of this in `GitLabInitialCommitAutomation`